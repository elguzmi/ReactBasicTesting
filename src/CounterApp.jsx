// rafc   = crear un functional component rapidamente
import { useState } from 'react'  // use significa hook , es una funcion especial
import PropTypes from 'prop-types'
 

// Los hooks son funciones
// eslint-disable-next-line react/prop-types
const CounterApp =({ counter })=>{  // la property no se cambia como tal
    const [ count  , setCount ] = useState( counter ); // desestructuracion   -> 0 es el valor del estado
    // fucniones flecha son mas rapidas y no cambian el valor del this
    const handleAdd = (event , message)=> {
        // cambio en el componenten y tiene que redibujar esos cambios, toca decirle a react que redibuje el functional component
        //setCount( count + 1); // decirle a react que el estado cambio y va a renderizar el componente
        setCount((c)=> c+ 1)
    }
    const handleSustract = ()=> setCount( count -1 );
    const handleReset = ()=> setCount(counter);
    return (
        <>
            <h1>Counter App</h1>
            <p>{ count }</p>
            <button  aria-label='btnSuma' onClick={ (event)=> handleAdd(event, 'Hola') }> +1 </button>
            <button onClick={ handleSustract }> -1 </button>
            <button aria-label='btn-reset' onClick={ handleReset }> reset </button>
        </>
    )
}


CounterApp.proptypes ={
    counter : PropTypes.number.isRequired
}
CounterApp.defaultProps={
    counter : 0
}

export default CounterApp