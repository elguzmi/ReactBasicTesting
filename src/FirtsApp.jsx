/*import { Fragment } from "react"  // se utiliza para evitar crear div padres
 <></>  significa sinonimo de fragmento
 {} expresion de javascript
 Un obj javascript no se puede colocar en una expresion
*/
import PropTypes from 'prop-types'  // definir el tipo a las properties

const getMessage = (a, b)=>{
    return a + b
}
//const name = 'Santiago Guzman'  // no va a re-renderizarlo cuando cambie de estado si esta afuera del functinal component
//const test = [1,2,3,4,5,6,7,8,9]
const textObj = {
    message: 'Hola Mundo',
    title : 'Fernando'
}
// eslint-disable-next-line react/prop-types
export const FirstApp = ({ title , subTitle = 'subtitulo', nombre})=> {
    const newMessage = getMessage(1,5)  
    //if(!title)throw new Error('No existe la propiedad title')

    return (
        <>  
            <h1 data-testid="test-title" > { title }  </h1>
            {/* <h1>{ title }</h1> */}
            <h3>{ subTitle }</h3>
            <p>{JSON.stringify(textObj)}</p>
            <p>{ getMessage(1,4)}</p>
            <p>{ newMessage }</p>
            <p>tu nombre es {nombre}</p>
        </>
    )
}


//**   Configuraciones de nuestro componentes **//
FirstApp.propTypes = {
    title : PropTypes.string.isRequired,
    subTitle: PropTypes.number
}

// los default propr entran antes que los proptypes
FirstApp.defaultProps = {
    subTitle: 2,
    nombre:'Fernando Herrera'

}