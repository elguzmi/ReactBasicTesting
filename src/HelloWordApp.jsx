// // todo en react comienzxa con un functional component (Son basadas en funciones los componentes)
// function App(){  //functional component
//     // document.createElement 
//     return (<h1>Hola Mundo desde react!!</h1>)
// }

// export default App


// functional componnets
const HelloWordApp = () => {
  return (
    <div>
      <h1>Hola Mundo</h1>
    </div>
  )
}

export default HelloWordApp

