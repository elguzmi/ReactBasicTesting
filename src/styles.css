html, body{
    background-color: rgb(30, 30, 30);
    color: rgb(214, 214, 214);
    font-family: system-ui, -apple-system, BlinkMacSystemFont, 'Segoe UI', Roboto, Oxygen, Ubuntu, Cantarell, 'Open Sans', 'Helvetica Neue', sans-serif;
    font-size: 1.3rem;
    text-align: center;
    font-weight: 400;
}

button{
    padding: 10px;
    font-size: 1.3rem;
    margin-right:  5px;
}