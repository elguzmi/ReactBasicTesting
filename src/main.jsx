//jsx significa que hay codigo xml = JS

import React from 'react'
import ReactDom from 'react-dom/client' // para renderizarlo
// import HelloWordApp from './HelloWordApp'
// import { FirstApp } from './FirtsApp'
import  CounterApp  from './CounterApp'
import './styles.css'
// import {App} from './HelloWordApp'



// la comilla sencilla '' es mas ligera que la dobel comilla ""
ReactDom.createRoot(document.getElementById('root')).render(
    <React.StrictMode>
        <CounterApp counter={12}></CounterApp>
        {/* <HelloWordApp/> */}
        {/* <FirstApp title="Hola Soy Vegeta" subTitle={123}></FirstApp>  */}
    </React.StrictMode>
)

