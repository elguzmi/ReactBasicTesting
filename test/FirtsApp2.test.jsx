import { render , screen } from "@testing-library/react"  // screen el objeto que estamos renderizando
import { FirstApp } from "../src/FirtsApp"

/* eslint-disable no-undef */
describe( 'Purebas en <FirtsApp>', ()=>{
    const title = 'Hola, Soy Goku' 
    const subtitle = '3' 
    test('debe de hacer math con el snaptshot ', () => {
      const { container } = render(<FirstApp title={title}></FirstApp>)
      expect( container ).toMatchSnapshot()
    })
    
    test('debe mostrar le mensaje "Hola soy goku" ', () => {
        render(<FirstApp title={title}></FirstApp>)
        //screen.debug() // muesta el componente que se renderiza
        expect(screen.getByText(title)).toBeTruthy()
    })

    test('debe de mostrar le titulo en un h1', () => {
       render(<FirstApp title={title}></FirstApp>)
       expect(screen.getByRole('heading' , {level : 1}).innerHTML ).toContain(title)
    })
    test('debe mostrar el subtitulo enviado por props', () => {
        render(<FirstApp title={title} subTitle={3}></FirstApp>);
        expect(screen.getAllByText(subtitle).length).toBe(1) 


    })
    
    
    
    
})