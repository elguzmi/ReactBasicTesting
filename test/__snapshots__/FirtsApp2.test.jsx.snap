// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Purebas en <FirtsApp> debe de hacer math con el snaptshot  1`] = `
<div>
  <h1
    data-testid="test-title"
  >
     
    Hola, Soy Goku
      
  </h1>
  <h3>
    2
  </h3>
  <p>
    {"message":"Hola Mundo","title":"Fernando"}
  </p>
  <p>
    5
  </p>
  <p>
    6
  </p>
  <p>
    tu nombre es 
    Fernando Herrera
  </p>
</div>
`;
